one_collect/src/perf_event/rb/source.rs

```diff
@@ -463,6 +463,24 @@ impl RingBufDataSource {
         Ok(())
     }
 
+    fn tasks_for_pids(pids: &mut Vec<i32>) {
+        let mut tasks = HashSet::new();
+
+        /* Find all unique tasks IDs */
+        for pid in pids.drain(..) {
+            tasks.insert(pid);
```

Comment:
If we insert `pid` here, then we're going to end up with duplicate pids when this function returns, no?

Reply:
You're right - I missed that it calls `drain` to empty them out.